lib/rails_icons/icon.rb

```diff
@@ -48,9 +52,7 @@ def default_attributes
   end
 
   def set
-    return @set if @set.present?
-
-    RailsIcons.configuration.default_set
+    @set.presence
```

Comment:
I did it like this because in `initialize` method if `set` is passed `nil`, it will set `@set` to `""` cause of `.to_s` call.
```ruby
  def initialize(name:, library:, args:, set: nil)
    @name = name
    @library = library.to_s
    @set = set.to_s
    @args = args
  end
``` 

in that case, in my implementation if `set` method is called it'll return `nil` if its a `""`, which indicates no set exists.

Initially I did just have `@set` as return value but if it returns `""`  it might effect any method call that expects response to be a string value having actual set name.

what do you think?

Reply:
updated the entire class to use `variant` instead of `set`. Also I'm explicitly using `RailsIcons.configuration.default_variant` if the library is `[:heroicons, :lucide, :tabler]` either one of these other wise I return `nil` because it can just have any variant at all.